### PR TITLE
Add Energy status property to room & creep logic changes

### PR DESCRIPTION
**Pull request by justonhart**

### CODE REVIEW

File: src/modules/memoryManagement.ts [hunk 1]
```diff
@@ -25,3 +25,22 @@ function handleDeadCreep(creepName: string) {
 
     delete Memory.creeps[creepName];
 }
+
+export function unclaimRoom(room: Room) {
+    if (room?.controller?.my) {
+        room.controller.unclaim();
+    }
+
+    if (room?.find(FIND_MY_CONSTRUCTION_SITES).length) {
```
Comment (lipkam) [inline src/modules/memoryManagement.ts:34]: Want to set a variable so you dont have to call find twice? If screeps caches the data it doesnt matter but not sure it does.

File: src/modules/populationManagement.ts [hunk 2]
```diff
@@ -116,40 +134,41 @@ function phaseTwoSpawning(spawn: StructureSpawn) {
...
     ) {
         options.memory.role = Role.BUILDER;
         partBlockToUse = WORKER_PART_BLOCK;
+    } else if (roomCreeps.filter((creep) => creep.memory.role === Role.UPGRADER).length < upgraderLimit) {
+        options.memory.role = Role.UPGRADER;
+        partBlockToUse = WORKER_PART_BLOCK;
+    } else if (roomCreeps.filter((creep) => creep.memory.role === Role.MAINTAINTER).length < maintainerLimit) {
+        options.memory.role = Role.MAINTAINTER;
+        partBlockToUse = WORKER_PART_BLOCK;
+        creepLevelCap = Math.ceil(spawn.room.energyCapacityAvailable / 200 / 2);
```
Comment (lipkam) [inline src/modules/populationManagement.ts:163]: why not just divide by 400? Shouldn't that be the same?

File: src/modules/memoryManagement.ts (hunk 1 above)
Comment (justonhart) [reply src/modules/memoryManagement.ts:34]: I'm pretty sure it does

File: src/modules/populationManagement.ts (hunk 2 above)
Comment (justonhart) [reply src/modules/populationManagement.ts:163]: I left it like that for the sake of logical readability, if that makes sense. 200 is the cost of [WORK, CARRY, MOVE], and halving that creates a creep of half size. Could probably make a constant or calculated value to take the place of 200 to that end, but this works for me.

Yes 400 works, but loses CONTEXT

**Review by lipkam — APPROVED** [review]

**Pull request merged**